File: pset6/credit/credit.py
import cs50


def main():
    number = int(get_creditcard())
    card = number
    # if card < pow(10, 13) or card >= pow(10, 16):
    #     print("INVALID")
    checksum = 0
    while card:
        pair = card % 100  # last two digit of the card
        p_digit = pair // 10
        a_digit = pair % 10
        p_digit *= 2
        if p_digit // 10 > 0:  # if true then add each digit to the sum
            pairsum = p_digit // 10 + p_digit % 10
            checksum = checksum + pairsum + a_digit
        else:
            pairsum = p_digit
            checksum = checksum + pairsum + a_digit
        card = card // 100
    if checksum % 10 == 0:  # check which company
        if number // pow(10, 13) == 34 or number // pow(10, 13) == 37:
            print("AMEX")
        elif 51 <= number // pow(10, 14) and number // pow(10, 14) <= 55:
            print("MASTERCARD")
        elif number // pow(10, 12) == 4 or number // pow(10, 15) == 4:
            print("VISA")
    else:
        print("INVALID")


def get_creditcard():
    while True:
        number = cs50.get_string("Enter credit card number:")
        if number.isnumeric() == True:
            break
    return number


main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    long number, checknumber;
    long pair;
    int p_digit, a_digit, pairsum, checksum = 0;
    do
    {
        number = get_long("Enter credit card number:");
        //Copy of number to check which credit card issuer
        printf("Number: %li\n", number);
        checknumber = number;
    }
    //When the number is less than 13-digit or bigger than 16-digit, go back to 'do-loop' to enter again.
    //When 'while (expr)' is true, go back to 'do' again; otherwise, break.
    while (!number);
    if (number < pow(10, 13) || number >= pow(10, 16))
    {
        printf("INVALID\n");
        //Exit code stays zero as the program terminates normally
        return 0;
    }
    while (number)
    {
        pair = number % 100;
        p_digit = pair / 10;
        a_digit = pair % 10;
        p_digit *= 2;
        if (p_digit / 10 > 0)
        {
            pairsum = p_digit / 10 + p_digit % 10;
            checksum = checksum + pairsum + a_digit;
        }
        else
        {
            pairsum = p_digit;
            checksum = checksum + pairsum + a_digit;
        }
        number /= 100;
    }
    if (checksum % 10 == 0)
    {
        if ((checknumber / (long)pow(10, 13) == 34) || (checknumber / (long)pow(10, 13) == 37))
        {
            printf("AMEX\n");
        }
        else if (51 <= (checknumber / (long)pow(10, 14)) && (checknumber / (long)pow(10, 14)) <= 55)
        {
            printf("MASTERCARD\n");
        }
        else if ((checknumber / (long)pow(10, 12)) == 4 || (checknumber / (long)pow(10, 15) == 4))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }

    }
    else
    {
        printf("INVALID\n");
    }
    return 0;